myserver.h

```diff
@@ -0,0 +1,69 @@
+#ifndef MYSERVER_H
+#define MYSERVER_H
+
+#include <QTcpServer>
+#include <QTcpSocket>
+#include <QJsonObject>
+#include <QJsonDocument>
+#include <QString>
+#include <QByteArray>
+#include <QtCore>
+#include <iostream>
+#include <QIODevice>
+#include <string>
+#include <vector>
+#include <memory.h>
+#include <QRegularExpression>
+#include <QStringConverter>
+#include "lib/json.hpp"
+
+#define PORT 5555
+#define TIME 5000
+
+class CMyServer: public QTcpServer{
+    Q_OBJECT
+private:
+    int iUserId = -1;
+    int iNoteId = -1;
+    int iFriendId = -1;
+    int iCategoryId = -1;
+    int iFrReqId = -1;
+    int iId = -1;
+    QTcpSocket* socket;
+    QByteArray Data;
+    QByteArray res;
+    QJsonDocument document;
+    QByteArray bytes;
+public:
+    CMyServer();
+    ~CMyServer();
+
+public slots:
+    void startServer();
+    void incomingConnection(qintptr pisocketDescriptor);
+    void sockReady();
+    void sockDisc();
+    void m_AddUser(std::vector <std::string> &msg);
```

Comment:
looks like this vector may be passed as const 
const std::vector <std::string> &msg

Reply:
in almost all the functions

Reply:
Fixed it